src/Flex.php

```diff
@@ -653,23 +654,23 @@ public function generateFlexId()
         $this->updateComposerLock();
     }
 
-    private function fetchRecipes(): array
+    public function fetchRecipes(array &$operations): array
```

Comment:
Is the by-reference necessary? It looks like it's just to clear `$this->operations` - could we just do that above after calling this method?

Reply:
Yes we can but i need to pass the operations in the new command and the reference improve the memory. Or, I can create a new method to define the operations ?

Reply:
>  the reference improve the memory.

Are you sure it does ? Do you have numbers proving it ? With PHP's copy-on-write behavior, such assumption about references is often wrong.

Reply:
I read this in the blackfire's book and i could be wrong about the use case. Can you explain my mistake ?

Reply:
I'd prefer code clarity over performance first. And then profile things and see where bottlenecks are. So, I would not pass as a reference, and then, after calling `$this->fetchRecipes($this->operations);` above, do `$this->operations = []`.